Reference: [BERT Github](https://github.com/google-research/bert); [Text Classification Tutorial](https://www.tensorflow.org/hub/tutorials/tf2_text_classification)

In [0]:
# Copyright 2019 Google Inc.

# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at

#     http://www.apache.org/licenses/LICENSE-2.0

# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# BBC News Classification with BERT on TF Hub

Part of the BBC News Classification [project](https://github.com/fzsea/Latest-News-Classifier/tree/nusfdu). Some code was adapted from [this colab notebook](https://colab.sandbox.google.com/github/google-research/bert/blob/master/predicting_movie_reviews_with_bert_on_tf_hub.ipynb).

In [0]:
from sklearn.model_selection import train_test_split
import pandas as pd
import tensorflow as tf
import tensorflow_hub as hub
from datetime import datetime

In [4]:
import bert
from bert import optimization
from bert import tokenization
from bert import run_classifier


Below, we'll set an output directory location to store our model output and checkpoints. This can be a local directory, in which case you'd set OUTPUT_DIR to the name of the directory you'd like to create. If you're running this code in Google's hosted Colab, the directory won't persist after the Colab session ends.

Alternatively, if you're a GCP user, you can store output in a GCP bucket. To do that, set a directory name in OUTPUT_DIR and the name of the GCP bucket in the BUCKET field.

Set DO_DELETE to rewrite the OUTPUT_DIR if it exists. Otherwise, Tensorflow will load existing model checkpoints from that directory (if they exist).

In [5]:
# Set the output directory for saving model file
# Optionally, set a GCP bucket location

OUTPUT_DIR = 'OUTPUT_DIR_NAME'#@param {type:"string"}
#@markdown Whether or not to clear/delete the directory and create a new one
DO_DELETE = False #@param {type:"boolean"}
#@markdown Set USE_BUCKET and BUCKET if you want to (optionally) store model output on GCP bucket.
USE_BUCKET = False #@param {type:"boolean"}
BUCKET = 'BUCKET_NAME' #@param {type:"string"}

if USE_BUCKET:
  OUTPUT_DIR = 'gs://{}/{}'.format(BUCKET, OUTPUT_DIR)
  from google.colab import auth
  auth.authenticate_user()

if DO_DELETE:
  try:
    tf.gfile.DeleteRecursively(OUTPUT_DIR)
  except:
    # Doesn't matter if the directory didn't exist
    pass
tf.gfile.MakeDirs(OUTPUT_DIR)
print('***** Model output directory: {} *****'.format(OUTPUT_DIR))


***** Model output directory: OUTPUT_DIR_NAME *****


#### Data

First, let's download the dataset, hosted on the GitHub.

In [0]:
import requests
import urllib.request

DATA_PATH = 'https://github.com/fzsea/Latest-News-Classifier/raw/nusfdu/0.%20Latest%20News%20Classifier/01.%20Dataset%20Creation/News_dataset.csv'
urllib.request.urlretrieve(DATA_PATH, 'News_dataset.csv')

raw_data = pd.read_csv('/content/News_dataset.csv', sep=';')

In [7]:
raw_data.head()

File_Name  ... Complete_Filename
0   001.txt  ...  001.txt-business
1   002.txt  ...  002.txt-business
2   003.txt  ...  003.txt-business
3   004.txt  ...  004.txt-business
4   005.txt  ...  005.txt-business

[5 rows x 4 columns]

In [0]:
category_codes = {
    'business': 0,
    'entertainment': 1,
    'politics': 2,
    'sport': 3,
    'tech': 4
}

raw_data['Category_Code'] = raw_data['Category']
raw_data = raw_data.replace({'Category_Code':category_codes})

In [0]:
train, test = train_test_split(raw_data,
                               test_size=0.15,
                               random_state=8)

To keep training fast, we'll take a sample of 5000 train and test examples, respectively.

In [10]:
train.columns

Index(['File_Name', 'Content', 'Category', 'Complete_Filename',
       'Category_Code'],
      dtype='object')

In [11]:
train.head()

File_Name  ... Category_Code
1881   058.txt  ...             4
1104   209.txt  ...             2
358    359.txt  ...             0
478    479.txt  ...             0
1063   168.txt  ...             2

[5 rows x 5 columns]

For us, our input data is the 'sentence' column and our label is the 'polarity' column (0, 1 for negative and positive, respecitvely)

In [0]:
DATA_COLUMN = 'Content'
LABEL_COLUMN = 'Category_Code'
# label_list is the list of labels, i.e. True, False or 0, 1 or 'dog', 'cat'
label_list = [0, 1, 2, 3, 4]

#Data Preprocessing
We'll need to transform our data into a format BERT understands. This involves two steps. First, we create  `InputExample`'s using the constructor provided in the BERT library.

- `text_a` is the text we want to classify, which in this case, is the `Request` field in our Dataframe. 
- `text_b` is used if we're training a model to understand the relationship between sentences (i.e. is `text_b` a translation of `text_a`? Is `text_b` an answer to the question asked by `text_a`?). This doesn't apply to our task, so we can leave `text_b` blank.
- `label` is the label for our example, i.e. True, False

In [0]:
# Use the InputExample class from BERT's run_classifier code to create examples from the data
train_InputExamples = train.apply(lambda x: bert.run_classifier.InputExample(guid=None, # Globally unique ID for bookkeeping, unused in this example
                                                                   text_a = x[DATA_COLUMN], 
                                                                   text_b = None, 
                                                                   label = x[LABEL_COLUMN]), axis = 1)

test_InputExamples = test.apply(lambda x: bert.run_classifier.InputExample(guid=None, 
                                                                   text_a = x[DATA_COLUMN], 
                                                                   text_b = None, 
                                                                   label = x[LABEL_COLUMN]), axis = 1)

In [14]:
type(train_InputExamples)

pandas.core.series.Series

In [15]:
train_InputExamples.loc(0)

In [16]:
train_InputExamples.head()

1881    <bert.run_classifier.InputExample object at 0x...
1104    <bert.run_classifier.InputExample object at 0x...
358     <bert.run_classifier.InputExample object at 0x...
478     <bert.run_classifier.InputExample object at 0x...
1063    <bert.run_classifier.InputExample object at 0x...
dtype: object

Next, we need to preprocess our data so that it matches the data BERT was trained on. For this, we'll need to do a couple of things (but don't worry--this is also included in the Python library):


1. Lowercase our text (if we're using a BERT lowercase model)
2. Tokenize it (i.e. "sally says hi" -> ["sally", "says", "hi"])
3. Break words into WordPieces (i.e. "calling" -> ["call", "##ing"])
4. Map our words to indexes using a vocab file that BERT provides
5. Add special "CLS" and "SEP" tokens (see the [readme](https://github.com/google-research/bert))
6. Append "index" and "segment" tokens to each input (see the [BERT paper](https://arxiv.org/pdf/1810.04805.pdf))

Happily, we don't have to worry about most of these details.




To start, we'll need to load a vocabulary file and lowercasing information directly from the BERT tf hub module:

In [17]:
# This is a path to an uncased (all lowercase) version of BERT
BERT_MODEL_HUB = "https://tfhub.dev/google/bert_uncased_L-12_H-768_A-12/1"

def create_tokenizer_from_hub_module():
  """Get the vocab file and casing info from the Hub module."""
  with tf.Graph().as_default():
    bert_module = hub.Module(BERT_MODEL_HUB)
    tokenization_info = bert_module(signature="tokenization_info", as_dict=True)
    with tf.Session() as sess:
      vocab_file, do_lower_case = sess.run([tokenization_info["vocab_file"],
                                            tokenization_info["do_lower_case"]])
      
  return bert.tokenization.FullTokenizer(
      vocab_file=vocab_file, do_lower_case=do_lower_case)

tokenizer = create_tokenizer_from_hub_module()

INFO:tensorflow:Saver not created because there are no variables in the graph to restore


INFO:tensorflow:Saver not created because there are no variables in the graph to restore


Great--we just learned that the BERT model we're using expects lowercase data (that's what stored in tokenization_info["do_lower_case"]) and we also loaded BERT's vocab file. We also created a tokenizer, which breaks words into word pieces:

In [18]:
tokenizer.tokenize("This here's an example of using the BERT tokenizer")

['this',
 'here',
 "'",
 's',
 'an',
 'example',
 'of',
 'using',
 'the',
 'bert',
 'token',
 '##izer']

Using our tokenizer, we'll call `run_classifier.convert_examples_to_features` on our InputExamples to convert them into features BERT understands.

In [30]:
# We'll set sequences to be at most 256 tokens long.
MAX_SEQ_LENGTH = 256
# Convert our train and test features to InputFeatures that BERT understands.
train_features = bert.run_classifier.convert_examples_to_features(train_InputExamples, label_list, MAX_SEQ_LENGTH, tokenizer)
test_features = bert.run_classifier.convert_examples_to_features(test_InputExamples, label_list, MAX_SEQ_LENGTH, tokenizer)

INFO:tensorflow:Writing example 0 of 1891


INFO:tensorflow:Writing example 0 of 1891


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] bt boost ##s its broadband packages british telecom has said it will double the broadband speeds of most of its home and business customers . the increased speeds will come at no extra charge and follows a similar move by internet service provider ao ##l . many bt customers will now have download speeds of 2 ##mb ##ps , although there are usage allowance ##s of between one gig ##aby ##te and 30 gig ##aby ##tes a month . the new speeds start to come into effect on 17 february for home customers and 1 april for businesses . " britain is now broadband britain , " said duncan ingram , bt ' s managing director , broadband and internet services . he added : " ninety percent of our customers will see real increases in speed . " these speed increases will give people the opportunity to do a lot more with their broadband connections , " he said . up ##load speeds - the speed at which information is sent from a pc via broadband - will remain at the same speed , said

INFO:tensorflow:tokens: [CLS] bt boost ##s its broadband packages british telecom has said it will double the broadband speeds of most of its home and business customers . the increased speeds will come at no extra charge and follows a similar move by internet service provider ao ##l . many bt customers will now have download speeds of 2 ##mb ##ps , although there are usage allowance ##s of between one gig ##aby ##te and 30 gig ##aby ##tes a month . the new speeds start to come into effect on 17 february for home customers and 1 april for businesses . " britain is now broadband britain , " said duncan ingram , bt ' s managing director , broadband and internet services . he added : " ninety percent of our customers will see real increases in speed . " these speed increases will give people the opportunity to do a lot more with their broadband connections , " he said . up ##load speeds - the speed at which information is sent from a pc via broadband - will remain at the same speed , said

INFO:tensorflow:input_ids: 101 18411 12992 2015 2049 19595 14555 2329 18126 2038 2056 2009 2097 3313 1996 19595 10898 1997 2087 1997 2049 2188 1998 2449 6304 1012 1996 3445 10898 2097 2272 2012 2053 4469 3715 1998 4076 1037 2714 2693 2011 4274 2326 10802 20118 2140 1012 2116 18411 6304 2097 2085 2031 8816 10898 1997 1016 14905 4523 1010 2348 2045 2024 8192 21447 2015 1997 2090 2028 15453 21275 2618 1998 2382 15453 21275 4570 1037 3204 1012 1996 2047 10898 2707 2000 2272 2046 3466 2006 2459 2337 2005 2188 6304 1998 1015 2258 2005 5661 1012 1000 3725 2003 2085 19595 3725 1010 1000 2056 7343 23231 1010 18411 1005 1055 6605 2472 1010 19595 1998 4274 2578 1012 2002 2794 1024 1000 13568 3867 1997 2256 6304 2097 2156 2613 7457 1999 3177 1012 1000 2122 3177 7457 2097 2507 2111 1996 4495 2000 2079 1037 2843 2062 2007 2037 19595 7264 1010 1000 2002 2056 1012 2039 11066 10898 1011 1996 3177 2012 2029 2592 2003 2741 2013 1037 7473 3081 19595 1011 2097 3961 2012 1996 2168 3177 1010 2056 2720 23231 

INFO:tensorflow:input_ids: 101 18411 12992 2015 2049 19595 14555 2329 18126 2038 2056 2009 2097 3313 1996 19595 10898 1997 2087 1997 2049 2188 1998 2449 6304 1012 1996 3445 10898 2097 2272 2012 2053 4469 3715 1998 4076 1037 2714 2693 2011 4274 2326 10802 20118 2140 1012 2116 18411 6304 2097 2085 2031 8816 10898 1997 1016 14905 4523 1010 2348 2045 2024 8192 21447 2015 1997 2090 2028 15453 21275 2618 1998 2382 15453 21275 4570 1037 3204 1012 1996 2047 10898 2707 2000 2272 2046 3466 2006 2459 2337 2005 2188 6304 1998 1015 2258 2005 5661 1012 1000 3725 2003 2085 19595 3725 1010 1000 2056 7343 23231 1010 18411 1005 1055 6605 2472 1010 19595 1998 4274 2578 1012 2002 2794 1024 1000 13568 3867 1997 2256 6304 2097 2156 2613 7457 1999 3177 1012 1000 2122 3177 7457 2097 2507 2111 1996 4495 2000 2079 1037 2843 2062 2007 2037 19595 7264 1010 1000 2002 2056 1012 2039 11066 10898 1011 1996 3177 2012 2029 2592 2003 2741 2013 1037 7473 3081 19595 1011 2097 3961 2012 1996 2168 3177 1010 2056 2720 23231 

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 4 (id = 4)


INFO:tensorflow:label: 4 (id = 4)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] abbas ' will not tolerate ' attacks palestinian leader mahmoud abbas has said he will not tolerate attacks such as last friday ' s suicide bombing in the israeli city of tel aviv . in an interview ahead of a meeting in london to discuss palestinian reforms , mr abbas said such attacks were against palestinian interests . the palestinian authority ( pa ) was ex ##ert ##ing " a 100 % effort " to end the violence , mr abbas added . the attack , which killed five , was the first of its kind since he took office . mr abbas confirmed israel shared information with the pa in the hunt for the organise ##rs of the attack . the israeli government refuses to accept syria ' s denial ##s that it was implicated in the nightclub bombing . israeli officials gave an intelligence briefing to foreign ambassadors on monday , explaining syria ' s alleged involvement . british foreign minister jack straw said there had been a " continuing stream " of information suggesting pale

INFO:tensorflow:tokens: [CLS] abbas ' will not tolerate ' attacks palestinian leader mahmoud abbas has said he will not tolerate attacks such as last friday ' s suicide bombing in the israeli city of tel aviv . in an interview ahead of a meeting in london to discuss palestinian reforms , mr abbas said such attacks were against palestinian interests . the palestinian authority ( pa ) was ex ##ert ##ing " a 100 % effort " to end the violence , mr abbas added . the attack , which killed five , was the first of its kind since he took office . mr abbas confirmed israel shared information with the pa in the hunt for the organise ##rs of the attack . the israeli government refuses to accept syria ' s denial ##s that it was implicated in the nightclub bombing . israeli officials gave an intelligence briefing to foreign ambassadors on monday , explaining syria ' s alleged involvement . british foreign minister jack straw said there had been a " continuing stream " of information suggesting pale

INFO:tensorflow:input_ids: 101 17532 1005 2097 2025 19242 1005 4491 9302 3003 27278 17532 2038 2056 2002 2097 2025 19242 4491 2107 2004 2197 5958 1005 1055 5920 8647 1999 1996 5611 2103 1997 10093 12724 1012 1999 2019 4357 3805 1997 1037 3116 1999 2414 2000 6848 9302 8818 1010 2720 17532 2056 2107 4491 2020 2114 9302 5426 1012 1996 9302 3691 1006 6643 1007 2001 4654 8743 2075 1000 1037 2531 1003 3947 1000 2000 2203 1996 4808 1010 2720 17532 2794 1012 1996 2886 1010 2029 2730 2274 1010 2001 1996 2034 1997 2049 2785 2144 2002 2165 2436 1012 2720 17532 4484 3956 4207 2592 2007 1996 6643 1999 1996 5690 2005 1996 22933 2869 1997 1996 2886 1012 1996 5611 2231 10220 2000 5138 7795 1005 1055 14920 2015 2008 2009 2001 20467 1999 1996 15479 8647 1012 5611 4584 2435 2019 4454 27918 2000 3097 20986 2006 6928 1010 9990 7795 1005 1055 6884 6624 1012 2329 3097 2704 2990 13137 2056 2045 2018 2042 1037 1000 5719 5460 1000 1997 2592 9104 9302 16830 2967 2020 4082 2013 2306 7795 1012 1999 2019 10373 4357

INFO:tensorflow:input_ids: 101 17532 1005 2097 2025 19242 1005 4491 9302 3003 27278 17532 2038 2056 2002 2097 2025 19242 4491 2107 2004 2197 5958 1005 1055 5920 8647 1999 1996 5611 2103 1997 10093 12724 1012 1999 2019 4357 3805 1997 1037 3116 1999 2414 2000 6848 9302 8818 1010 2720 17532 2056 2107 4491 2020 2114 9302 5426 1012 1996 9302 3691 1006 6643 1007 2001 4654 8743 2075 1000 1037 2531 1003 3947 1000 2000 2203 1996 4808 1010 2720 17532 2794 1012 1996 2886 1010 2029 2730 2274 1010 2001 1996 2034 1997 2049 2785 2144 2002 2165 2436 1012 2720 17532 4484 3956 4207 2592 2007 1996 6643 1999 1996 5690 2005 1996 22933 2869 1997 1996 2886 1012 1996 5611 2231 10220 2000 5138 7795 1005 1055 14920 2015 2008 2009 2001 20467 1999 1996 15479 8647 1012 5611 4584 2435 2019 4454 27918 2000 3097 20986 2006 6928 1010 9990 7795 1005 1055 6884 6624 1012 2329 3097 2704 2990 13137 2056 2045 2018 2042 1037 1000 5719 5460 1000 1997 2592 9104 9302 16830 2967 2020 4082 2013 2306 7795 1012 1999 2019 10373 4357

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 2 (id = 2)


INFO:tensorflow:label: 2 (id = 2)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] gm in crunch talks on fiat future fiat will meet car giant general motors ( gm ) on tuesday in an attempt to reach agreement over the future of the italian firm ' s loss - making auto group . fiat claims that gm is legally obliged to buy the 90 % of the car unit it does not already own ; gm says the contract , signed in 2000 , is no longer valid . press reports have speculated that fiat may be willing to accept a cash payment in return for dropping its claim . both companies want to cut costs as the car industry adjust ##s to wan ##ing demand . the meeting between fiat boss sergio march ##ion ##ne and gm ' s rick wagon ##er is due to take place at 133 ##0 gm ##t in zurich , according to the reuters news agency . mr march ##ion ##ne is confident of his firm ' s legal position , saying in an interview with the financial times that gm ' s argument " has no legs " . the agreement in question dates back to gm ' s decision to buy 20 % of fiat ' s auto division i

INFO:tensorflow:tokens: [CLS] gm in crunch talks on fiat future fiat will meet car giant general motors ( gm ) on tuesday in an attempt to reach agreement over the future of the italian firm ' s loss - making auto group . fiat claims that gm is legally obliged to buy the 90 % of the car unit it does not already own ; gm says the contract , signed in 2000 , is no longer valid . press reports have speculated that fiat may be willing to accept a cash payment in return for dropping its claim . both companies want to cut costs as the car industry adjust ##s to wan ##ing demand . the meeting between fiat boss sergio march ##ion ##ne and gm ' s rick wagon ##er is due to take place at 133 ##0 gm ##t in zurich , according to the reuters news agency . mr march ##ion ##ne is confident of his firm ' s legal position , saying in an interview with the financial times that gm ' s argument " has no legs " . the agreement in question dates back to gm ' s decision to buy 20 % of fiat ' s auto division i

INFO:tensorflow:input_ids: 101 13938 1999 24514 7566 2006 18550 2925 18550 2097 3113 2482 5016 2236 9693 1006 13938 1007 2006 9857 1999 2019 3535 2000 3362 3820 2058 1996 2925 1997 1996 3059 3813 1005 1055 3279 1011 2437 8285 2177 1012 18550 4447 2008 13938 2003 10142 14723 2000 4965 1996 3938 1003 1997 1996 2482 3131 2009 2515 2025 2525 2219 1025 13938 2758 1996 3206 1010 2772 1999 2456 1010 2003 2053 2936 9398 1012 2811 4311 2031 15520 2008 18550 2089 2022 5627 2000 5138 1037 5356 7909 1999 2709 2005 7510 2049 4366 1012 2119 3316 2215 2000 3013 5366 2004 1996 2482 3068 14171 2015 2000 14071 2075 5157 1012 1996 3116 2090 18550 5795 13983 2233 3258 2638 1998 13938 1005 1055 6174 9540 2121 2003 2349 2000 2202 2173 2012 14506 2692 13938 2102 1999 10204 1010 2429 2000 1996 26665 2739 4034 1012 2720 2233 3258 2638 2003 9657 1997 2010 3813 1005 1055 3423 2597 1010 3038 1999 2019 4357 2007 1996 3361 2335 2008 13938 1005 1055 6685 1000 2038 2053 3456 1000 1012 1996 3820 1999 3160 5246 2067 20

INFO:tensorflow:input_ids: 101 13938 1999 24514 7566 2006 18550 2925 18550 2097 3113 2482 5016 2236 9693 1006 13938 1007 2006 9857 1999 2019 3535 2000 3362 3820 2058 1996 2925 1997 1996 3059 3813 1005 1055 3279 1011 2437 8285 2177 1012 18550 4447 2008 13938 2003 10142 14723 2000 4965 1996 3938 1003 1997 1996 2482 3131 2009 2515 2025 2525 2219 1025 13938 2758 1996 3206 1010 2772 1999 2456 1010 2003 2053 2936 9398 1012 2811 4311 2031 15520 2008 18550 2089 2022 5627 2000 5138 1037 5356 7909 1999 2709 2005 7510 2049 4366 1012 2119 3316 2215 2000 3013 5366 2004 1996 2482 3068 14171 2015 2000 14071 2075 5157 1012 1996 3116 2090 18550 5795 13983 2233 3258 2638 1998 13938 1005 1055 6174 9540 2121 2003 2349 2000 2202 2173 2012 14506 2692 13938 2102 1999 10204 1010 2429 2000 1996 26665 2739 4034 1012 2720 2233 3258 2638 2003 9657 1997 2010 3813 1005 1055 3423 2597 1010 3038 1999 2019 4357 2007 1996 3361 2335 2008 13938 1005 1055 6685 1000 2038 2053 3456 1000 1012 1996 3820 1999 3160 5246 2067 20

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] ford gains from finance not cars ford , the us car company , reported higher fourth quarter and full - year profits on thursday boosted by a bu ##oya ##nt period for its car loans unit . net income for 2004 was $ 3 . 5 ##bn ( a ##£ ##1 . 87 ##bn ) - up nearly $ 3 ##bn from 2003 - while turnover rose $ 7 . 2 ##bn to $ 170 . 8 ##bn . in the fourth quarter alone ford reported net income of $ 104 ##m , compared with a loss of $ 79 ##3 ##m a year ago . but its auto unit made a loss . fourth quarter turnover was $ 44 . 7 ##bn , compared to $ 45 . 9 ##bn a year ago . though car and truck loan profits saved the day , ford ' s auto unit made a pre - tax loss of $ 470 ##m in the fourth quarter ( compared to a profit of a ##£ ##13 ##m in the year - ago period ) and its us sales dipped 3 . 8 % . yesterday general motor ' s results also showed its finance unit was a strong contributor to profits . however , ford is working hard to rev ##ital ##ise its product portfolio

INFO:tensorflow:tokens: [CLS] ford gains from finance not cars ford , the us car company , reported higher fourth quarter and full - year profits on thursday boosted by a bu ##oya ##nt period for its car loans unit . net income for 2004 was $ 3 . 5 ##bn ( a ##£ ##1 . 87 ##bn ) - up nearly $ 3 ##bn from 2003 - while turnover rose $ 7 . 2 ##bn to $ 170 . 8 ##bn . in the fourth quarter alone ford reported net income of $ 104 ##m , compared with a loss of $ 79 ##3 ##m a year ago . but its auto unit made a loss . fourth quarter turnover was $ 44 . 7 ##bn , compared to $ 45 . 9 ##bn a year ago . though car and truck loan profits saved the day , ford ' s auto unit made a pre - tax loss of $ 470 ##m in the fourth quarter ( compared to a profit of a ##£ ##13 ##m in the year - ago period ) and its us sales dipped 3 . 8 % . yesterday general motor ' s results also showed its finance unit was a strong contributor to profits . however , ford is working hard to rev ##ital ##ise its product portfolio

INFO:tensorflow:input_ids: 101 4811 12154 2013 5446 2025 3765 4811 1010 1996 2149 2482 2194 1010 2988 3020 2959 4284 1998 2440 1011 2095 11372 2006 9432 28043 2011 1037 20934 18232 3372 2558 2005 2049 2482 10940 3131 1012 5658 3318 2005 2432 2001 1002 1017 1012 1019 24700 1006 1037 29646 2487 1012 6584 24700 1007 1011 2039 3053 1002 1017 24700 2013 2494 1011 2096 20991 3123 1002 1021 1012 1016 24700 2000 1002 10894 1012 1022 24700 1012 1999 1996 2959 4284 2894 4811 2988 5658 3318 1997 1002 9645 2213 1010 4102 2007 1037 3279 1997 1002 6535 2509 2213 1037 2095 3283 1012 2021 2049 8285 3131 2081 1037 3279 1012 2959 4284 20991 2001 1002 4008 1012 1021 24700 1010 4102 2000 1002 3429 1012 1023 24700 1037 2095 3283 1012 2295 2482 1998 4744 5414 11372 5552 1996 2154 1010 4811 1005 1055 8285 3131 2081 1037 3653 1011 4171 3279 1997 1002 21064 2213 1999 1996 2959 4284 1006 4102 2000 1037 5618 1997 1037 29646 17134 2213 1999 1996 2095 1011 3283 2558 1007 1998 2049 2149 4341 13537 1017 1012 1022 10

INFO:tensorflow:input_ids: 101 4811 12154 2013 5446 2025 3765 4811 1010 1996 2149 2482 2194 1010 2988 3020 2959 4284 1998 2440 1011 2095 11372 2006 9432 28043 2011 1037 20934 18232 3372 2558 2005 2049 2482 10940 3131 1012 5658 3318 2005 2432 2001 1002 1017 1012 1019 24700 1006 1037 29646 2487 1012 6584 24700 1007 1011 2039 3053 1002 1017 24700 2013 2494 1011 2096 20991 3123 1002 1021 1012 1016 24700 2000 1002 10894 1012 1022 24700 1012 1999 1996 2959 4284 2894 4811 2988 5658 3318 1997 1002 9645 2213 1010 4102 2007 1037 3279 1997 1002 6535 2509 2213 1037 2095 3283 1012 2021 2049 8285 3131 2081 1037 3279 1012 2959 4284 20991 2001 1002 4008 1012 1021 24700 1010 4102 2000 1002 3429 1012 1023 24700 1037 2095 3283 1012 2295 2482 1998 4744 5414 11372 5552 1996 2154 1010 4811 1005 1055 8285 3131 2081 1037 3653 1011 4171 3279 1997 1002 21064 2213 1999 1996 2959 4284 1006 4102 2000 1037 5618 1997 1037 29646 17134 2213 1999 1996 2095 1011 3283 2558 1007 1998 2049 2149 4341 13537 1017 1012 1022 10

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] mps ' murder sentence concern murder sentences should not be reduced automatically simply because of a guilty plea , says a new mps ' report . the influential commons home affairs committee was responding to sentencing guidelines issued this summer . the mps also call for tough ##er sentences for crimes committed under the influence of drink or drugs . they say the influence of drugs and alcohol should be introduced as an ag ##gra ##vating factor when judges and magistrates sentence offenders . committee chairman john den ##ham said drugs of alcohol were sometimes used as an excuse . " the committee believes that these arguments should be rejected by sentence ##rs and that being under their influence should instead be an ag ##gra ##vating factor . " at present judges , when sentencing murderers to the mandatory life sentence , can reduce the tariff - the minimum term they must serve - if the defendant plead ##s guilty . but although they are spared the ord

INFO:tensorflow:tokens: [CLS] mps ' murder sentence concern murder sentences should not be reduced automatically simply because of a guilty plea , says a new mps ' report . the influential commons home affairs committee was responding to sentencing guidelines issued this summer . the mps also call for tough ##er sentences for crimes committed under the influence of drink or drugs . they say the influence of drugs and alcohol should be introduced as an ag ##gra ##vating factor when judges and magistrates sentence offenders . committee chairman john den ##ham said drugs of alcohol were sometimes used as an excuse . " the committee believes that these arguments should be rejected by sentence ##rs and that being under their influence should instead be an ag ##gra ##vating factor . " at present judges , when sentencing murderers to the mandatory life sentence , can reduce the tariff - the minimum term they must serve - if the defendant plead ##s guilty . but although they are spared the ord

INFO:tensorflow:input_ids: 101 12616 1005 4028 6251 5142 4028 11746 2323 2025 2022 4359 8073 3432 2138 1997 1037 5905 14865 1010 2758 1037 2047 12616 1005 3189 1012 1996 6383 7674 2188 3821 2837 2001 14120 2000 23280 11594 3843 2023 2621 1012 1996 12616 2036 2655 2005 7823 2121 11746 2005 6997 5462 2104 1996 3747 1997 4392 2030 5850 1012 2027 2360 1996 3747 1997 5850 1998 6544 2323 2022 3107 2004 2019 12943 17643 26477 5387 2043 6794 1998 23007 6251 19591 1012 2837 3472 2198 7939 3511 2056 5850 1997 6544 2020 2823 2109 2004 2019 8016 1012 1000 1996 2837 7164 2008 2122 9918 2323 2022 5837 2011 6251 2869 1998 2008 2108 2104 2037 3747 2323 2612 2022 2019 12943 17643 26477 5387 1012 1000 2012 2556 6794 1010 2043 23280 28882 2000 1996 10915 2166 6251 1010 2064 5547 1996 23234 1011 1996 6263 2744 2027 2442 3710 1011 2065 1996 13474 25803 2015 5905 1012 2021 2348 2027 2024 16891 1996 23304 1997 1037 3979 2116 4028 5694 1005 9064 2024 12511 1012 1999 2251 2023 2095 8282 3410 1005 1055 6359 101

INFO:tensorflow:input_ids: 101 12616 1005 4028 6251 5142 4028 11746 2323 2025 2022 4359 8073 3432 2138 1997 1037 5905 14865 1010 2758 1037 2047 12616 1005 3189 1012 1996 6383 7674 2188 3821 2837 2001 14120 2000 23280 11594 3843 2023 2621 1012 1996 12616 2036 2655 2005 7823 2121 11746 2005 6997 5462 2104 1996 3747 1997 4392 2030 5850 1012 2027 2360 1996 3747 1997 5850 1998 6544 2323 2022 3107 2004 2019 12943 17643 26477 5387 2043 6794 1998 23007 6251 19591 1012 2837 3472 2198 7939 3511 2056 5850 1997 6544 2020 2823 2109 2004 2019 8016 1012 1000 1996 2837 7164 2008 2122 9918 2323 2022 5837 2011 6251 2869 1998 2008 2108 2104 2037 3747 2323 2612 2022 2019 12943 17643 26477 5387 1012 1000 2012 2556 6794 1010 2043 23280 28882 2000 1996 10915 2166 6251 1010 2064 5547 1996 23234 1011 1996 6263 2744 2027 2442 3710 1011 2065 1996 13474 25803 2015 5905 1012 2021 2348 2027 2024 16891 1996 23304 1997 1037 3979 2116 4028 5694 1005 9064 2024 12511 1012 1999 2251 2023 2095 8282 3410 1005 1055 6359 101

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 2 (id = 2)


INFO:tensorflow:label: 2 (id = 2)


INFO:tensorflow:Writing example 0 of 334


INFO:tensorflow:Writing example 0 of 334


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] ireland call up un ##cap ##ped campbell ulster sc ##rum - half ki ##eran campbell is one of five un ##cap ##ped players included in ireland ' s rb ##s six nations squad . campbell is joined by ulster colleagues roger wilson and ronan mcc ##or ##mack along with connacht ' s bernard jack ##man and munster ' s shaun payne . gordon d ' arc ##y is back after injury while munster flank ##er alan quinlan also returns to international consideration . " the squad is selected purely on form . a lot of players put their hands up , " coach eddie o ' sullivan told bbc sport . " ki ##eran campbell was just one of those players . he has been playing very well in the he ##ine ##ken cup and deserves his call - up . " there is big competition in some departments and not so much in others . there were one or two players who were unfortunate just to miss out . " back - row forwards david wallace and victor costello are omitted , with o ' sullivan having quinlan , wilson , sim

INFO:tensorflow:tokens: [CLS] ireland call up un ##cap ##ped campbell ulster sc ##rum - half ki ##eran campbell is one of five un ##cap ##ped players included in ireland ' s rb ##s six nations squad . campbell is joined by ulster colleagues roger wilson and ronan mcc ##or ##mack along with connacht ' s bernard jack ##man and munster ' s shaun payne . gordon d ' arc ##y is back after injury while munster flank ##er alan quinlan also returns to international consideration . " the squad is selected purely on form . a lot of players put their hands up , " coach eddie o ' sullivan told bbc sport . " ki ##eran campbell was just one of those players . he has been playing very well in the he ##ine ##ken cup and deserves his call - up . " there is big competition in some departments and not so much in others . there were one or two players who were unfortunate just to miss out . " back - row forwards david wallace and victor costello are omitted , with o ' sullivan having quinlan , wilson , sim

INFO:tensorflow:input_ids: 101 3163 2655 2039 4895 17695 5669 6063 11059 8040 6824 1011 2431 11382 23169 6063 2003 2028 1997 2274 4895 17695 5669 2867 2443 1999 3163 1005 1055 21144 2015 2416 3741 4686 1012 6063 2003 2587 2011 11059 8628 5074 4267 1998 18633 23680 2953 26945 2247 2007 27062 1005 1055 6795 2990 2386 1998 11348 1005 1055 16845 13470 1012 5146 1040 1005 8115 2100 2003 2067 2044 4544 2096 11348 12205 2121 5070 28451 2036 5651 2000 2248 9584 1012 1000 1996 4686 2003 3479 11850 2006 2433 1012 1037 2843 1997 2867 2404 2037 2398 2039 1010 1000 2873 5752 1051 1005 7624 2409 4035 4368 1012 1000 11382 23169 6063 2001 2074 2028 1997 2216 2867 1012 2002 2038 2042 2652 2200 2092 1999 1996 2002 3170 7520 2452 1998 17210 2010 2655 1011 2039 1012 1000 2045 2003 2502 2971 1999 2070 7640 1998 2025 2061 2172 1999 2500 1012 2045 2020 2028 2030 2048 2867 2040 2020 15140 2074 2000 3335 2041 1012 1000 2067 1011 5216 19390 2585 7825 1998 5125 21015 2024 16647 1010 2007 1051 1005 7624 2383 2845

INFO:tensorflow:input_ids: 101 3163 2655 2039 4895 17695 5669 6063 11059 8040 6824 1011 2431 11382 23169 6063 2003 2028 1997 2274 4895 17695 5669 2867 2443 1999 3163 1005 1055 21144 2015 2416 3741 4686 1012 6063 2003 2587 2011 11059 8628 5074 4267 1998 18633 23680 2953 26945 2247 2007 27062 1005 1055 6795 2990 2386 1998 11348 1005 1055 16845 13470 1012 5146 1040 1005 8115 2100 2003 2067 2044 4544 2096 11348 12205 2121 5070 28451 2036 5651 2000 2248 9584 1012 1000 1996 4686 2003 3479 11850 2006 2433 1012 1037 2843 1997 2867 2404 2037 2398 2039 1010 1000 2873 5752 1051 1005 7624 2409 4035 4368 1012 1000 11382 23169 6063 2001 2074 2028 1997 2216 2867 1012 2002 2038 2042 2652 2200 2092 1999 1996 2002 3170 7520 2452 1998 17210 2010 2655 1011 2039 1012 1000 2045 2003 2502 2971 1999 2070 7640 1998 2025 2061 2172 1999 2500 1012 2045 2020 2028 2030 2048 2867 2040 2020 15140 2074 2000 3335 2041 1012 1000 2067 1011 5216 19390 2585 7825 1998 5125 21015 2024 16647 1010 2007 1051 1005 7624 2383 2845

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 3 (id = 3)


INFO:tensorflow:label: 3 (id = 3)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] gu ##rk ##has to help tsunami victims britain has offered to send a company of 120 gu ##rk ##has to assist with the tsunami relief effort in indonesia , downing street said . the deployment would involve troops from the 2nd battalion royal gu ##rk ##ha rifles , based in brunei . discussions have begun with indonesia on the exact timing and location of the deployment , but the government said the offer was aimed at the ace ##h province . downing st said a similar offer might be made to the sri lankan government . however a spokesman pointed out that there were particular logistical difficulties in indonesia which the gu ##rk ##has might be able to help with . the spokesman said : " following this morning ' s daily coordination meeting on the post - tsunami relief effort , the government has formally offered the indonesian government the assistance of a company of british army gu ##rk ##has from 2nd battalion royal gu ##rk ##ha rifles around 120 personnel an

INFO:tensorflow:tokens: [CLS] gu ##rk ##has to help tsunami victims britain has offered to send a company of 120 gu ##rk ##has to assist with the tsunami relief effort in indonesia , downing street said . the deployment would involve troops from the 2nd battalion royal gu ##rk ##ha rifles , based in brunei . discussions have begun with indonesia on the exact timing and location of the deployment , but the government said the offer was aimed at the ace ##h province . downing st said a similar offer might be made to the sri lankan government . however a spokesman pointed out that there were particular logistical difficulties in indonesia which the gu ##rk ##has might be able to help with . the spokesman said : " following this morning ' s daily coordination meeting on the post - tsunami relief effort , the government has formally offered the indonesian government the assistance of a company of british army gu ##rk ##has from 2nd battalion royal gu ##rk ##ha rifles around 120 personnel an

INFO:tensorflow:input_ids: 101 19739 8024 14949 2000 2393 19267 5694 3725 2038 3253 2000 4604 1037 2194 1997 6036 19739 8024 14949 2000 6509 2007 1996 19267 4335 3947 1999 6239 1010 22501 2395 2056 1012 1996 10813 2052 9125 3629 2013 1996 3416 4123 2548 19739 8024 3270 9494 1010 2241 1999 18692 1012 10287 2031 5625 2007 6239 2006 1996 6635 10984 1998 3295 1997 1996 10813 1010 2021 1996 2231 2056 1996 3749 2001 6461 2012 1996 9078 2232 2874 1012 22501 2358 2056 1037 2714 3749 2453 2022 2081 2000 1996 5185 16159 2231 1012 2174 1037 14056 4197 2041 2008 2045 2020 3327 28961 8190 1999 6239 2029 1996 19739 8024 14949 2453 2022 2583 2000 2393 2007 1012 1996 14056 2056 1024 1000 2206 2023 2851 1005 1055 3679 12016 3116 2006 1996 2695 1011 19267 4335 3947 1010 1996 2231 2038 6246 3253 1996 9003 2231 1996 5375 1997 1037 2194 1997 2329 2390 19739 8024 14949 2013 3416 4123 2548 19739 8024 3270 9494 2105 6036 5073 1998 2048 12400 1012 1000 2023 2003 1999 2804 2000 1996 3719 1998 2948 2057 2031 252

INFO:tensorflow:input_ids: 101 19739 8024 14949 2000 2393 19267 5694 3725 2038 3253 2000 4604 1037 2194 1997 6036 19739 8024 14949 2000 6509 2007 1996 19267 4335 3947 1999 6239 1010 22501 2395 2056 1012 1996 10813 2052 9125 3629 2013 1996 3416 4123 2548 19739 8024 3270 9494 1010 2241 1999 18692 1012 10287 2031 5625 2007 6239 2006 1996 6635 10984 1998 3295 1997 1996 10813 1010 2021 1996 2231 2056 1996 3749 2001 6461 2012 1996 9078 2232 2874 1012 22501 2358 2056 1037 2714 3749 2453 2022 2081 2000 1996 5185 16159 2231 1012 2174 1037 14056 4197 2041 2008 2045 2020 3327 28961 8190 1999 6239 2029 1996 19739 8024 14949 2453 2022 2583 2000 2393 2007 1012 1996 14056 2056 1024 1000 2206 2023 2851 1005 1055 3679 12016 3116 2006 1996 2695 1011 19267 4335 3947 1010 1996 2231 2038 6246 3253 1996 9003 2231 1996 5375 1997 1037 2194 1997 2329 2390 19739 8024 14949 2013 3416 4123 2548 19739 8024 3270 9494 2105 6036 5073 1998 2048 12400 1012 1000 2023 2003 1999 2804 2000 1996 3719 1998 2948 2057 2031 252

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 2 (id = 2)


INFO:tensorflow:label: 2 (id = 2)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] egypt and israel seal trade deal in a sign of a tha ##w in relations between egypt and israel , the two countries have signed a trade protocol with the us , allowing egyptian goods made in partnership with israeli firms free access to american markets . the protocol , signed in cairo , will establish what are called " qualified industrial zones " in egypt . products from these zones will enjoy duty free access to the us , provided that 35 % of their components are the product of israeli - egyptian cooperation . the us describes this as the most important economic agreement between egypt and israel in two decades . the protocol establishing the zones has been stalled for years . there has been deep sensitivity in egypt about any form of co - operation with israel as long as its peace process with the palestinians remains blocked . but in recent weeks an unusual warmth has crept into relations between the two countries . both exchanged prisoners earlier this

INFO:tensorflow:tokens: [CLS] egypt and israel seal trade deal in a sign of a tha ##w in relations between egypt and israel , the two countries have signed a trade protocol with the us , allowing egyptian goods made in partnership with israeli firms free access to american markets . the protocol , signed in cairo , will establish what are called " qualified industrial zones " in egypt . products from these zones will enjoy duty free access to the us , provided that 35 % of their components are the product of israeli - egyptian cooperation . the us describes this as the most important economic agreement between egypt and israel in two decades . the protocol establishing the zones has been stalled for years . there has been deep sensitivity in egypt about any form of co - operation with israel as long as its peace process with the palestinians remains blocked . but in recent weeks an unusual warmth has crept into relations between the two countries . both exchanged prisoners earlier this

INFO:tensorflow:input_ids: 101 5279 1998 3956 7744 3119 3066 1999 1037 3696 1997 1037 22794 2860 1999 4262 2090 5279 1998 3956 1010 1996 2048 3032 2031 2772 1037 3119 8778 2007 1996 2149 1010 4352 6811 5350 2081 1999 5386 2007 5611 9786 2489 3229 2000 2137 6089 1012 1996 8778 1010 2772 1999 11096 1010 2097 5323 2054 2024 2170 1000 4591 3919 10019 1000 1999 5279 1012 3688 2013 2122 10019 2097 5959 4611 2489 3229 2000 1996 2149 1010 3024 2008 3486 1003 1997 2037 6177 2024 1996 4031 1997 5611 1011 6811 6792 1012 1996 2149 5577 2023 2004 1996 2087 2590 3171 3820 2090 5279 1998 3956 1999 2048 5109 1012 1996 8778 7411 1996 10019 2038 2042 20659 2005 2086 1012 2045 2038 2042 2784 14639 1999 5279 2055 2151 2433 1997 2522 1011 3169 2007 3956 2004 2146 2004 2049 3521 2832 2007 1996 21524 3464 8534 1012 2021 1999 3522 3134 2019 5866 8251 2038 13147 2046 4262 2090 1996 2048 3032 1012 2119 10573 5895 3041 2023 3204 1010 2007 5279 13041 2067 2019 5611 2040 2038 2366 2809 2086 1999 3827 2044 2108 797

INFO:tensorflow:input_ids: 101 5279 1998 3956 7744 3119 3066 1999 1037 3696 1997 1037 22794 2860 1999 4262 2090 5279 1998 3956 1010 1996 2048 3032 2031 2772 1037 3119 8778 2007 1996 2149 1010 4352 6811 5350 2081 1999 5386 2007 5611 9786 2489 3229 2000 2137 6089 1012 1996 8778 1010 2772 1999 11096 1010 2097 5323 2054 2024 2170 1000 4591 3919 10019 1000 1999 5279 1012 3688 2013 2122 10019 2097 5959 4611 2489 3229 2000 1996 2149 1010 3024 2008 3486 1003 1997 2037 6177 2024 1996 4031 1997 5611 1011 6811 6792 1012 1996 2149 5577 2023 2004 1996 2087 2590 3171 3820 2090 5279 1998 3956 1999 2048 5109 1012 1996 8778 7411 1996 10019 2038 2042 20659 2005 2086 1012 2045 2038 2042 2784 14639 1999 5279 2055 2151 2433 1997 2522 1011 3169 2007 3956 2004 2146 2004 2049 3521 2832 2007 1996 21524 3464 8534 1012 2021 1999 3522 3134 2019 5866 8251 2038 13147 2046 4262 2090 1996 2048 3032 1012 2119 10573 5895 3041 2023 3204 1010 2007 5279 13041 2067 2019 5611 2040 2038 2366 2809 2086 1999 3827 2044 2108 797

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] cai ##rn shares up on new oil find shares in cai ##rn energy have jumped 6 % after the firm said an indian oil ##field was larger than previously thought . cai ##rn said drilling to the north - west of its development site in rajasthan had produced " very strong results " . the company also said it now believed the development area would be able to produce oil for more than 25 years . cai ##rn ' s share price rose 300 % last year after a number of oil finds , but its shares were hit in december following a disappointing drilling update . december ' s share fall means that cai ##rn is still in danger of being relegated from the ft ##se 100 when the index is res ##hu ##ffled next month . cai ##rn ' s shares closed up 64 pen ##ce , or 6 % , at 113 ##0 ##p on thursday . before christmas , cai ##rn revealed that drilling to the north of the field in rajasthan had been disappointing , which caused its shares to lose 18 % in one day . however , on thursday , the 

INFO:tensorflow:tokens: [CLS] cai ##rn shares up on new oil find shares in cai ##rn energy have jumped 6 % after the firm said an indian oil ##field was larger than previously thought . cai ##rn said drilling to the north - west of its development site in rajasthan had produced " very strong results " . the company also said it now believed the development area would be able to produce oil for more than 25 years . cai ##rn ' s share price rose 300 % last year after a number of oil finds , but its shares were hit in december following a disappointing drilling update . december ' s share fall means that cai ##rn is still in danger of being relegated from the ft ##se 100 when the index is res ##hu ##ffled next month . cai ##rn ' s shares closed up 64 pen ##ce , or 6 % , at 113 ##0 ##p on thursday . before christmas , cai ##rn revealed that drilling to the north of the field in rajasthan had been disappointing , which caused its shares to lose 18 % in one day . however , on thursday , the 

INFO:tensorflow:input_ids: 101 29080 6826 6661 2039 2006 2047 3514 2424 6661 1999 29080 6826 2943 2031 5598 1020 1003 2044 1996 3813 2056 2019 2796 3514 3790 2001 3469 2084 3130 2245 1012 29080 6826 2056 15827 2000 1996 2167 1011 2225 1997 2049 2458 2609 1999 16815 2018 2550 1000 2200 2844 3463 1000 1012 1996 2194 2036 2056 2009 2085 3373 1996 2458 2181 2052 2022 2583 2000 3965 3514 2005 2062 2084 2423 2086 1012 29080 6826 1005 1055 3745 3976 3123 3998 1003 2197 2095 2044 1037 2193 1997 3514 4858 1010 2021 2049 6661 2020 2718 1999 2285 2206 1037 15640 15827 10651 1012 2285 1005 1055 3745 2991 2965 2008 29080 6826 2003 2145 1999 5473 1997 2108 7049 2013 1996 3027 3366 2531 2043 1996 5950 2003 24501 6979 28579 2279 3204 1012 29080 6826 1005 1055 6661 2701 2039 4185 7279 3401 1010 2030 1020 1003 1010 2012 12104 2692 2361 2006 9432 1012 2077 4234 1010 29080 6826 3936 2008 15827 2000 1996 2167 1997 1996 2492 1999 16815 2018 2042 15640 1010 2029 3303 2049 6661 2000 4558 2324 1003 1999 2028 2

INFO:tensorflow:input_ids: 101 29080 6826 6661 2039 2006 2047 3514 2424 6661 1999 29080 6826 2943 2031 5598 1020 1003 2044 1996 3813 2056 2019 2796 3514 3790 2001 3469 2084 3130 2245 1012 29080 6826 2056 15827 2000 1996 2167 1011 2225 1997 2049 2458 2609 1999 16815 2018 2550 1000 2200 2844 3463 1000 1012 1996 2194 2036 2056 2009 2085 3373 1996 2458 2181 2052 2022 2583 2000 3965 3514 2005 2062 2084 2423 2086 1012 29080 6826 1005 1055 3745 3976 3123 3998 1003 2197 2095 2044 1037 2193 1997 3514 4858 1010 2021 2049 6661 2020 2718 1999 2285 2206 1037 15640 15827 10651 1012 2285 1005 1055 3745 2991 2965 2008 29080 6826 2003 2145 1999 5473 1997 2108 7049 2013 1996 3027 3366 2531 2043 1996 5950 2003 24501 6979 28579 2279 3204 1012 29080 6826 1005 1055 6661 2701 2039 4185 7279 3401 1010 2030 1020 1003 1010 2012 12104 2692 2361 2006 9432 1012 2077 4234 1010 29080 6826 3936 2008 15827 2000 1996 2167 1997 1996 2492 1999 16815 2018 2042 15640 1010 2029 3303 2049 6661 2000 4558 2324 1003 1999 2028 2

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:*** Example ***


INFO:tensorflow:*** Example ***


INFO:tensorflow:guid: None


INFO:tensorflow:guid: None


INFO:tensorflow:tokens: [CLS] saudi nc ##ci ' s shares so ##ar shares in saudi arabia ' s national company for cooperative insurance ( nc ##ci ) soared on their first day of trading in ri ##yad ##h . they were trading 84 % above the offer price on monday , changing hands at 37 ##2 ri ##yal ##s ( $ 99 ; a ##£ ##53 ) after topping 400 early in the day . demand for the ins ##urer ' s debut shares was strong - 12 times what was on sale . the listing was part of the country ' s plans to open up its insurance market and boost demand in the sector . der ##eg ##ulation is expected to boost demand for accident and damage cover . previously , only nc ##ci has been legally allowed to offer insurance products within saudi arabia . however , the authorities have turned a blind eye to the many other firms selling insurance . saudi arabia now wants a fully functioning insurance industry and is introducing legislation that will cl ##amp down on una ##uth ##oris ##ed companies . policy - makers also wa

INFO:tensorflow:tokens: [CLS] saudi nc ##ci ' s shares so ##ar shares in saudi arabia ' s national company for cooperative insurance ( nc ##ci ) soared on their first day of trading in ri ##yad ##h . they were trading 84 % above the offer price on monday , changing hands at 37 ##2 ri ##yal ##s ( $ 99 ; a ##£ ##53 ) after topping 400 early in the day . demand for the ins ##urer ' s debut shares was strong - 12 times what was on sale . the listing was part of the country ' s plans to open up its insurance market and boost demand in the sector . der ##eg ##ulation is expected to boost demand for accident and damage cover . previously , only nc ##ci has been legally allowed to offer insurance products within saudi arabia . however , the authorities have turned a blind eye to the many other firms selling insurance . saudi arabia now wants a fully functioning insurance industry and is introducing legislation that will cl ##amp down on una ##uth ##oris ##ed companies . policy - makers also wa

INFO:tensorflow:input_ids: 101 8174 13316 6895 1005 1055 6661 2061 2906 6661 1999 8174 9264 1005 1055 2120 2194 2005 10791 5427 1006 13316 6895 1007 29127 2006 2037 2034 2154 1997 6202 1999 15544 25152 2232 1012 2027 2020 6202 6391 1003 2682 1996 3749 3976 2006 6928 1010 5278 2398 2012 4261 2475 15544 21095 2015 1006 1002 5585 1025 1037 29646 22275 1007 2044 22286 4278 2220 1999 1996 2154 1012 5157 2005 1996 16021 27595 1005 1055 2834 6661 2001 2844 1011 2260 2335 2054 2001 2006 5096 1012 1996 10328 2001 2112 1997 1996 2406 1005 1055 3488 2000 2330 2039 2049 5427 3006 1998 12992 5157 1999 1996 4753 1012 4315 13910 9513 2003 3517 2000 12992 5157 2005 4926 1998 4053 3104 1012 3130 1010 2069 13316 6895 2038 2042 10142 3039 2000 3749 5427 3688 2306 8174 9264 1012 2174 1010 1996 4614 2031 2357 1037 6397 3239 2000 1996 2116 2060 9786 4855 5427 1012 8174 9264 2085 4122 1037 3929 12285 5427 3068 1998 2003 10449 6094 2008 2097 18856 16613 2091 2006 14477 14317 21239 2098 3316 1012 3343 1011 111

INFO:tensorflow:input_ids: 101 8174 13316 6895 1005 1055 6661 2061 2906 6661 1999 8174 9264 1005 1055 2120 2194 2005 10791 5427 1006 13316 6895 1007 29127 2006 2037 2034 2154 1997 6202 1999 15544 25152 2232 1012 2027 2020 6202 6391 1003 2682 1996 3749 3976 2006 6928 1010 5278 2398 2012 4261 2475 15544 21095 2015 1006 1002 5585 1025 1037 29646 22275 1007 2044 22286 4278 2220 1999 1996 2154 1012 5157 2005 1996 16021 27595 1005 1055 2834 6661 2001 2844 1011 2260 2335 2054 2001 2006 5096 1012 1996 10328 2001 2112 1997 1996 2406 1005 1055 3488 2000 2330 2039 2049 5427 3006 1998 12992 5157 1999 1996 4753 1012 4315 13910 9513 2003 3517 2000 12992 5157 2005 4926 1998 4053 3104 1012 3130 1010 2069 13316 6895 2038 2042 10142 3039 2000 3749 5427 3688 2306 8174 9264 1012 2174 1010 1996 4614 2031 2357 1037 6397 3239 2000 1996 2116 2060 9786 4855 5427 1012 8174 9264 2085 4122 1037 3929 12285 5427 3068 1998 2003 10449 6094 2008 2097 18856 16613 2091 2006 14477 14317 21239 2098 3316 1012 3343 1011 111

INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:input_mask: 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:segment_ids: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


INFO:tensorflow:label: 0 (id = 0)


INFO:tensorflow:label: 0 (id = 0)


In [20]:
type(train_features)

list

In [21]:
print(train_features[0])

#Creating a model

Now that we've prepared our data, let's focus on building a model. `create_model` does just this below. First, it loads the BERT tf hub module again (this time to extract the computation graph). Next, it creates a single new layer that will be trained to adapt BERT to our sentiment task (i.e. classifying whether a movie review is positive or negative). This strategy of using a mostly trained model is called [fine-tuning](http://wiki.fast.ai/index.php/Fine_tuning).

In [0]:
def create_model(is_predicting, input_ids, input_mask, segment_ids, labels,
                 num_labels):
  """Creates a classification model."""

  bert_module = hub.Module(
      BERT_MODEL_HUB,
      trainable=True)
  bert_inputs = dict(
      input_ids=input_ids,
      input_mask=input_mask,
      segment_ids=segment_ids)
  bert_outputs = bert_module(
      inputs=bert_inputs,
      signature="tokens",
      as_dict=True)

  # Use "pooled_output" for classification tasks on an entire sentence.
  # Use "sequence_outputs" for token-level output.
  output_layer = bert_outputs["pooled_output"]

  hidden_size = output_layer.shape[-1].value

  # Create our own layer to tune for politeness data.
  output_weights = tf.get_variable(
      "output_weights", [num_labels, hidden_size],
      initializer=tf.truncated_normal_initializer(stddev=0.02))

  output_bias = tf.get_variable(
      "output_bias", [num_labels], initializer=tf.zeros_initializer())

  with tf.variable_scope("loss"):

    # Dropout helps prevent overfitting
    output_layer = tf.nn.dropout(output_layer, keep_prob=0.9)

    logits = tf.matmul(output_layer, output_weights, transpose_b=True)
    logits = tf.nn.bias_add(logits, output_bias)
    log_probs = tf.nn.log_softmax(logits, axis=-1)

    # Convert labels into one-hot encoding
    one_hot_labels = tf.one_hot(labels, depth=num_labels, dtype=tf.float32)

    predicted_labels = tf.squeeze(tf.argmax(log_probs, axis=-1, output_type=tf.int32))
    # If we're predicting, we want predicted labels and the probabiltiies.
    if is_predicting:
      return (predicted_labels, log_probs)

    # If we're train/eval, compute loss between predicted and actual label
    per_example_loss = -tf.reduce_sum(one_hot_labels * log_probs, axis=-1)
    loss = tf.reduce_mean(per_example_loss)
    return (loss, predicted_labels, log_probs)


Next we'll wrap our model function in a `model_fn_builder` function that adapts our model to work for training, evaluation, and prediction.

Note: Turn off the metrics for the multi-class cases.

In [0]:
# model_fn_builder actually creates our model function
# using the passed parameters for num_labels, learning_rate, etc.
def model_fn_builder(num_labels, learning_rate, num_train_steps,
                     num_warmup_steps):
  """Returns `model_fn` closure for TPUEstimator."""
  def model_fn(features, labels, mode, params):  # pylint: disable=unused-argument
    """The `model_fn` for TPUEstimator."""

    input_ids = features["input_ids"]
    input_mask = features["input_mask"]
    segment_ids = features["segment_ids"]
    label_ids = features["label_ids"]

    is_predicting = (mode == tf.estimator.ModeKeys.PREDICT)
    
    # TRAIN and EVAL
    if not is_predicting:

      (loss, predicted_labels, log_probs) = create_model(
        is_predicting, input_ids, input_mask, segment_ids, label_ids, num_labels)

      train_op = bert.optimization.create_optimizer(
          loss, learning_rate, num_train_steps, num_warmup_steps, use_tpu=False)

      # Calculate evaluation metrics. 
      def metric_fn(label_ids, predicted_labels):
        accuracy = tf.metrics.accuracy(label_ids, predicted_labels)
        # f1_score = tf.contrib.metrics.f1_score(
        #     label_ids,
        #     predicted_labels)
        # auc = tf.metrics.auc(
        #     label_ids,
        #     predicted_labels)
        # recall = tf.metrics.recall(
        #     label_ids,
        #     predicted_labels)
        # precision = tf.metrics.precision(
        #     label_ids,
        #     predicted_labels) 
        # true_pos = tf.metrics.true_positives(
        #     label_ids,
        #     predicted_labels)
        # true_neg = tf.metrics.true_negatives(
        #     label_ids,
        #     predicted_labels)   
        # false_pos = tf.metrics.false_positives(
        #     label_ids,
        #     predicted_labels)  
        # false_neg = tf.metrics.false_negatives(
        #     label_ids,
        #     predicted_labels)
        return {
            "eval_accuracy": accuracy##,
            # "f1_score": f1_score,
            # "auc": auc,
            # "precision": precision,
            # "recall": recall,
            # "true_positives": true_pos,
            # "true_negatives": true_neg,
            # "false_positives": false_pos,
            # "false_negatives": false_neg
        }

      eval_metrics = metric_fn(label_ids, predicted_labels)

      if mode == tf.estimator.ModeKeys.TRAIN:
        return tf.estimator.EstimatorSpec(mode=mode,
          loss=loss,
          train_op=train_op)
      else:
          return tf.estimator.EstimatorSpec(mode=mode,
            loss=loss,
            eval_metric_ops=eval_metrics)
    else:
      (predicted_labels, log_probs) = create_model(
        is_predicting, input_ids, input_mask, segment_ids, label_ids, num_labels)

      predictions = {
          'probabilities': log_probs,
          'labels': predicted_labels
      }
      return tf.estimator.EstimatorSpec(mode, predictions=predictions)

  # Return the actual model function in the closure
  return model_fn


In [0]:
# Compute train and warmup steps from batch size
# These hyperparameters are copied from this colab notebook (https://colab.sandbox.google.com/github/tensorflow/tpu/blob/master/tools/colab/bert_finetuning_with_cloud_tpus.ipynb)
BATCH_SIZE = 32
LEARNING_RATE = 2e-5
NUM_TRAIN_EPOCHS = 3.0
# Warmup is a period of time where hte learning rate 
# is small and gradually increases--usually helps training.
WARMUP_PROPORTION = 0.1
# Model configs
SAVE_CHECKPOINTS_STEPS = 500
SAVE_SUMMARY_STEPS = 100

In [0]:
# Compute # train and warmup steps from batch size
num_train_steps = int(len(train_features) / BATCH_SIZE * NUM_TRAIN_EPOCHS)
num_warmup_steps = int(num_train_steps * WARMUP_PROPORTION)

In [0]:
# Specify outpit directory and number of checkpoint steps to save
run_config = tf.estimator.RunConfig(
    model_dir=OUTPUT_DIR,
    save_summary_steps=SAVE_SUMMARY_STEPS,
    save_checkpoints_steps=SAVE_CHECKPOINTS_STEPS)

In [43]:
model_fn = model_fn_builder(
  num_labels=len(label_list),
  learning_rate=LEARNING_RATE,
  num_train_steps=num_train_steps,
  num_warmup_steps=num_warmup_steps)

estimator = tf.estimator.Estimator(
  model_fn=model_fn,
  config=run_config,
  params={"batch_size": BATCH_SIZE})


INFO:tensorflow:Using config: {'_model_dir': 'OUTPUT_DIR_NAME', '_tf_random_seed': None, '_save_summary_steps': 100, '_save_checkpoints_steps': 500, '_save_checkpoints_secs': None, '_session_config': allow_soft_placement: true
graph_options {
  rewrite_options {
    meta_optimizer_iterations: ONE
  }
}
, '_keep_checkpoint_max': 5, '_keep_checkpoint_every_n_hours': 10000, '_log_step_count_steps': 100, '_train_distribute': None, '_device_fn': None, '_protocol': None, '_eval_distribute': None, '_experimental_distribute': None, '_experimental_max_worker_delay_secs': None, '_session_creation_timeout_secs': 7200, '_service': None, '_cluster_spec': <tensorflow.python.training.server_lib.ClusterSpec object at 0x7f86bf5540b8>, '_task_type': 'worker', '_task_id': 0, '_global_id_in_cluster': 0, '_master': '', '_evaluation_master': '', '_is_chief': True, '_num_ps_replicas': 0, '_num_worker_replicas': 1}


INFO:tensorflow:Using config: {'_model_dir': 'OUTPUT_DIR_NAME', '_tf_random_seed': None, '_save_summary_steps': 100, '_save_checkpoints_steps': 500, '_save_checkpoints_secs': None, '_session_config': allow_soft_placement: true
graph_options {
  rewrite_options {
    meta_optimizer_iterations: ONE
  }
}
, '_keep_checkpoint_max': 5, '_keep_checkpoint_every_n_hours': 10000, '_log_step_count_steps': 100, '_train_distribute': None, '_device_fn': None, '_protocol': None, '_eval_distribute': None, '_experimental_distribute': None, '_experimental_max_worker_delay_secs': None, '_session_creation_timeout_secs': 7200, '_service': None, '_cluster_spec': <tensorflow.python.training.server_lib.ClusterSpec object at 0x7f86bf5540b8>, '_task_type': 'worker', '_task_id': 0, '_global_id_in_cluster': 0, '_master': '', '_evaluation_master': '', '_is_chief': True, '_num_ps_replicas': 0, '_num_worker_replicas': 1}


Next we create an input builder function that takes our training feature set (`train_features`) and produces a generator. This is a pretty standard design pattern for working with Tensorflow [Estimators](https://www.tensorflow.org/guide/estimators).

In [0]:
# Create an input function for training. drop_remainder = True for using TPUs.
train_input_fn = bert.run_classifier.input_fn_builder(
    features=train_features,
    seq_length=MAX_SEQ_LENGTH,
    is_training=True,
    drop_remainder=False)

Now we train our model! For me, using a Colab notebook running on Google's GPUs, my training time was about 14 minutes.

In [38]:
print(f'Beginning Training!')
current_time = datetime.now()
estimator.train(input_fn=train_input_fn, max_steps=num_train_steps)
print("Training took time ", datetime.now() - current_time)

Beginning Training!
INFO:tensorflow:Calling model_fn.


INFO:tensorflow:Calling model_fn.


INFO:tensorflow:Saver not created because there are no variables in the graph to restore


INFO:tensorflow:Saver not created because there are no variables in the graph to restore
/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/indexed_slices.py:424: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "


INFO:tensorflow:Done calling model_fn.


INFO:tensorflow:Done calling model_fn.


INFO:tensorflow:Create CheckpointSaverHook.


INFO:tensorflow:Create CheckpointSaverHook.


INFO:tensorflow:Graph was finalized.


INFO:tensorflow:Graph was finalized.


INFO:tensorflow:Restoring parameters from OUTPUT_DIR_NAME/model.ckpt-0


INFO:tensorflow:Restoring parameters from OUTPUT_DIR_NAME/model.ckpt-0


Instructions for updating:
Use standard file utilities to get mtimes.


Instructions for updating:
Use standard file utilities to get mtimes.


INFO:tensorflow:Running local_init_op.


INFO:tensorflow:Running local_init_op.


INFO:tensorflow:Done running local_init_op.


INFO:tensorflow:Done running local_init_op.


INFO:tensorflow:Saving checkpoints for 0 into OUTPUT_DIR_NAME/model.ckpt.


INFO:tensorflow:Saving checkpoints for 0 into OUTPUT_DIR_NAME/model.ckpt.


INFO:tensorflow:loss = 1.6222804, step = 0


INFO:tensorflow:loss = 1.6222804, step = 0


INFO:tensorflow:global_step/sec: 0.541533


INFO:tensorflow:global_step/sec: 0.541533


INFO:tensorflow:loss = 0.07749912, step = 100 (184.662 sec)


INFO:tensorflow:loss = 0.07749912, step = 100 (184.662 sec)


INFO:tensorflow:Saving checkpoints for 177 into OUTPUT_DIR_NAME/model.ckpt.


INFO:tensorflow:Saving checkpoints for 177 into OUTPUT_DIR_NAME/model.ckpt.


INFO:tensorflow:Loss for final step: 0.0049373796.


INFO:tensorflow:Loss for final step: 0.0049373796.


Training took time  0:06:02.958145


Now let's use our test data to see how well our model did:

##### Training Accuracy

In [45]:
# Create an input function for training. drop_remainder = True for using TPUs.
eval_train_input_fn = bert.run_classifier.input_fn_builder(
    features=train_features,
    seq_length=MAX_SEQ_LENGTH,
    is_training=False,
    drop_remainder=False)

estimator.evaluate(input_fn=eval_train_input_fn, steps=None)

INFO:tensorflow:Calling model_fn.


INFO:tensorflow:Calling model_fn.


INFO:tensorflow:Saver not created because there are no variables in the graph to restore


INFO:tensorflow:Saver not created because there are no variables in the graph to restore
/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/indexed_slices.py:424: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "


INFO:tensorflow:Done calling model_fn.


INFO:tensorflow:Done calling model_fn.


INFO:tensorflow:Starting evaluation at 2020-06-10T23:16:09Z


INFO:tensorflow:Starting evaluation at 2020-06-10T23:16:09Z


INFO:tensorflow:Graph was finalized.


INFO:tensorflow:Graph was finalized.


INFO:tensorflow:Restoring parameters from OUTPUT_DIR_NAME/model.ckpt-177


INFO:tensorflow:Restoring parameters from OUTPUT_DIR_NAME/model.ckpt-177


INFO:tensorflow:Running local_init_op.


INFO:tensorflow:Running local_init_op.


INFO:tensorflow:Done running local_init_op.


INFO:tensorflow:Done running local_init_op.


INFO:tensorflow:Finished evaluation at 2020-06-10-23:16:48


INFO:tensorflow:Finished evaluation at 2020-06-10-23:16:48


INFO:tensorflow:Saving dict for global step 177: eval_accuracy = 0.9989424, global_step = 177, loss = 0.0074916948


INFO:tensorflow:Saving dict for global step 177: eval_accuracy = 0.9989424, global_step = 177, loss = 0.0074916948


INFO:tensorflow:Saving 'checkpoint_path' summary for global step 177: OUTPUT_DIR_NAME/model.ckpt-177


INFO:tensorflow:Saving 'checkpoint_path' summary for global step 177: OUTPUT_DIR_NAME/model.ckpt-177


{'eval_accuracy': 0.9989424, 'global_step': 177, 'loss': 0.0074916948}

##### Test Accuracy

In [0]:
test_input_fn = run_classifier.input_fn_builder(
    features=test_features,
    seq_length=MAX_SEQ_LENGTH,
    is_training=False,
    drop_remainder=False)

In [44]:
estimator.evaluate(input_fn=test_input_fn, steps=None)

INFO:tensorflow:Calling model_fn.


INFO:tensorflow:Calling model_fn.


INFO:tensorflow:Saver not created because there are no variables in the graph to restore


INFO:tensorflow:Saver not created because there are no variables in the graph to restore
/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/indexed_slices.py:424: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "


INFO:tensorflow:Done calling model_fn.


INFO:tensorflow:Done calling model_fn.


INFO:tensorflow:Starting evaluation at 2020-06-10T23:11:21Z


INFO:tensorflow:Starting evaluation at 2020-06-10T23:11:21Z


INFO:tensorflow:Graph was finalized.


INFO:tensorflow:Graph was finalized.


INFO:tensorflow:Restoring parameters from OUTPUT_DIR_NAME/model.ckpt-177


INFO:tensorflow:Restoring parameters from OUTPUT_DIR_NAME/model.ckpt-177


INFO:tensorflow:Running local_init_op.


INFO:tensorflow:Running local_init_op.


INFO:tensorflow:Done running local_init_op.


INFO:tensorflow:Done running local_init_op.


INFO:tensorflow:Finished evaluation at 2020-06-10-23:11:33


INFO:tensorflow:Finished evaluation at 2020-06-10-23:11:33


INFO:tensorflow:Saving dict for global step 177: eval_accuracy = 0.98203593, global_step = 177, loss = 0.039279792


INFO:tensorflow:Saving dict for global step 177: eval_accuracy = 0.98203593, global_step = 177, loss = 0.039279792


INFO:tensorflow:Saving 'checkpoint_path' summary for global step 177: OUTPUT_DIR_NAME/model.ckpt-177


INFO:tensorflow:Saving 'checkpoint_path' summary for global step 177: OUTPUT_DIR_NAME/model.ckpt-177


{'eval_accuracy': 0.98203593, 'global_step': 177, 'loss': 0.039279792}

Training result:  
``{'eval_accuracy': 0.9989424, 'global_step': 177, 'loss': 0.0074916948}``

Testing result:  
``{'eval_accuracy': 0.98203593, 'global_step': 177, 'loss': 0.039279792}``


Visualize the predictions:

In [55]:
test_pred_generator = estimator.predict(input_fn=test_input_fn)
test_pred_result = list(test_pred_generator)


INFO:tensorflow:Calling model_fn.


INFO:tensorflow:Calling model_fn.


INFO:tensorflow:Saver not created because there are no variables in the graph to restore


INFO:tensorflow:Saver not created because there are no variables in the graph to restore


INFO:tensorflow:Done calling model_fn.


INFO:tensorflow:Done calling model_fn.


INFO:tensorflow:Graph was finalized.


INFO:tensorflow:Graph was finalized.


INFO:tensorflow:Restoring parameters from OUTPUT_DIR_NAME/model.ckpt-177


INFO:tensorflow:Restoring parameters from OUTPUT_DIR_NAME/model.ckpt-177


INFO:tensorflow:Running local_init_op.


INFO:tensorflow:Running local_init_op.


INFO:tensorflow:Done running local_init_op.


INFO:tensorflow:Done running local_init_op.


In [56]:
test_pred_result[:5]

[{'labels': 3,
  'probabilities': array([-7.1887426e+00, -6.5005150e+00, -6.7207894e+00, -4.4607422e-03,
         -6.9203591e+00], dtype=float32)},
 {'labels': 2,
  'probabilities': array([-5.032318 , -6.8917365, -0.0098919, -6.638263 , -6.9137273],
        dtype=float32)},
 {'labels': 0,
  'probabilities': array([-4.1497555e-03, -7.1691031e+00, -6.3769398e+00, -7.2736192e+00,
         -6.9307895e+00], dtype=float32)},
 {'labels': 0,
  'probabilities': array([-5.4481360e-03, -6.8471060e+00, -6.4099474e+00, -6.6095715e+00,
         -6.5869465e+00], dtype=float32)},
 {'labels': 0,
  'probabilities': array([-3.7806004e-03, -6.9856677e+00, -6.9201832e+00, -6.9427691e+00,
         -7.0185180e+00], dtype=float32)}]

In [61]:
# category_codes = {
#     'business': 0,
#     'entertainment': 1,
#     'politics': 2,
#     'sport': 3,
#     'tech': 4
# }

code_2_category = dict([(v, k) for k, v in category_codes.items()])

cnt = 0
for text, label, res in zip(test['Content'], test['Category'], test_pred_result):
  print('Prediction: {}, True Label: {} \n\n {}'.format(code_2_category[res['labels']], label, text))
  print('###'*5 + '\n')
  cnt += 1
  if cnt == 5:
    break

Prediction: sport, True Label: sport 

 Ireland call up uncapped Campbell

Ulster scrum-half Kieran Campbell is one of five uncapped players included in Ireland's RBS Six Nations squad.

Campbell is joined by Ulster colleagues Roger Wilson and Ronan McCormack along with Connacht's Bernard Jackman and Munster's Shaun Payne. Gordon D'Arcy is back after injury while Munster flanker Alan Quinlan also returns to international consideration. "The squad is selected purely on form. A lot of players put their hands up," coach Eddie O'Sullivan told BBC Sport. "Kieran Campbell was just one of those players. He has been playing very well in the Heineken Cup and deserves his call-up. "There is big competition in some departments and not so much in others. There were one or two players who were unfortunate just to miss out." Back-row forwards David Wallace and Victor Costello are omitted, with O'Sullivan having Quinlan, Wilson, Simon Easterby, Anthony Foley, Denis Leamy and Johnny O'Connor vying for

In [62]:
### Focus on the wrong cases:
cnt = 0
for text, label, res in zip(test['Content'], test['Category'], test_pred_result):
  pred_label = code_2_category[res['labels']]
  if pred_label == label:
    continue
  print('Prediction: {}, True Label: {} \n\n {}'.format(pred_label, label, text))
  print('###'*5 + '\n')
  cnt += 1
  if cnt == 5:
    break

Prediction: business, True Label: politics 

 How political squabbles snowball

It's become commonplace to argue that Blair and Brown are like squabbling school kids and that they (and their supporters) need to grow up and stop bickering.

But this analysis in fact gets it wrong. It's not just children who fight - adults do too. And there are solid reasons why even a trivial argument between mature protagonists can be hard to stop once its got going. The key feature of an endless feud is that everyone can agree they'd be better off if it ended - but everyone wants to have the last word.

Each participant genuinely wants the row to stop, but thinks it worth prolonging the argument just a tiny bit to ensure their view is heard. Their successive attempts to end the argument with their last word ensure the argument goes on and on and on. (In the case of Mr Blair and Mr Brown, successive books are published, ensuring the issues never die.) Now this isn't because the participants are stupid 